src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/SuppressFormattingRule.cs

```diff
@@ -372,13 +374,13 @@ private void AddInitializerSuppressOperations(List<SuppressOperation> list, Synt
             // array or collection initializer case
             if (node.IsInitializerForArrayOrCollectionCreationExpression())
             {
-                var arrayOrCollectionInitializer = node as InitializerExpressionSyntax;
+                var arrayOrCollectionInitializer = (InitializerExpressionSyntax)node;
                 AddSuppressAllOperationIfOnMultipleLine(list, arrayOrCollectionInitializer.OpenBraceToken.GetPreviousToken(includeZeroWidth: true), arrayOrCollectionInitializer.CloseBraceToken);
                 return;
             }
 
             var initializer = GetInitializerNode(node);
-            if (initializer != null)
+            if (initializer is { Parent: { } })
```

Comment:
would honestly prefer `initializer?.Parent != null`

Reply:
or, wait on `initializer is { Parent: not null }`.  I find the `{ }` pattern for null checking to just be very lacking in clarity.  i.e. it looks like a bug or a does-nothing, andi have to switch my mind into peeling back a layer to figure out what it's really trying to convey.

Reply:
We'll have an analyzer and fix to convert `{ }` to `not null` where applicable, so I'm OK leaving this one for now.